pkg/asset/installconfig/azure/validation.go

```diff
@@ -521,19 +523,47 @@ func validateNetworks(client API, p *aztypes.Platform, machineNetworks []types.M
 			return append(allErrs, field.Invalid(fieldPath.Child("virtualNetwork"), p.VirtualNetwork, err.Error()))
 		}
 
-		computeSubnet, err := client.GetComputeSubnet(context.TODO(), p.NetworkResourceGroupName, p.VirtualNetwork, p.ComputeSubnet)
+		computeSubnetName := p.ComputeSubnet
+		controlPlaneSubnetName := p.ControlPlaneSubnet
+		if computeSubnetName == "" || controlPlaneSubnetName == "" {
+			for _, subnet := range p.Subnets {
+				if subnet.Role == capz.SubnetControlPlane && controlPlaneSubnetName == "" {
```

Comment:
This code is using the 1st subnet with ControlPlane and Compute roles as the respective subnet and ignoring the rest. Is that intentional? If so, could you add a comment indicating that and why?